packages/blueprints-integration/src/previews.ts

```diff
@@ -19,6 +23,55 @@ export enum PreviewType {
 	BlueprintImage = 'blueprintImage',
 }
 
+// The PreviewContent types are a partly replica of the types in PreviewPopUpContext.tsx
+export type PreviewContent =
+	| {
+			type: 'iframe'
+			href: string
+			postMessage?: any
+			dimensions?: { width: number; height: number }
+	  }
+	| {
+			type: 'image'
+			src: string
+	  }
+	| {
+			type: 'video'
+			src: string
+	  }
+	| {
+			type: 'script'
+			script?: string
+			firstWords?: string
+			lastWords?: string
+			comment?: string
+			lastModified?: number
+	  }
+	| {
+			type: 'title'
+			content: string
+	  }
+	| {
+			type: 'inOutWords'
+			in?: string
+			out: string
+	  }
+	| {
+			type: 'layerInfo'
+			layerType: SourceLayerType
+			text: Array<string>
+			inTime?: number | string
+			outTime?: number | string
+			duration?: number | string
+	  }
+	| {
+			type: 'separationLine'
+	  }
+	| {
+			type: 'data'
+			content: { key: string; value: string }[]
+	  }
```

Comment:
This interface specifically is very vague for a public-use interface. I think we need to make sure that it's more clear what one can expect/put into here.

Reply:
Could you clarify what additional specificity you'd like for the data type?
We should definitely dig into what this does and add some documentation, currently I don't know how this feature works, or where it's being used.

Reply:
Comments for it's purpose has been added to code.